core-bundle/src/Resources/contao/library/Contao/Database/Result.php

```diff
@@ -155,7 +153,17 @@ public function __get($strKey)
 				break;
 
 			case 'numFields':
-				return $this->resResult->columnCount();
+				if ($this->resResult)
+				{
+					return $this->resResult->columnCount();
+				}
+
+				if (isset($this->resultSet[0]))
```

Comment:
Now that you have removed the array initialization, this condition will never be executed, will it?

Reply:
It will be executed if you access `numFields` after all rows are fetched. In this case `resResult` would habe been set to `null`, see line 413.

Reply:
I see. 👍